app/controllers/claims_controller.rb

```diff
@@ -21,7 +22,15 @@ def create
 
   def show
     search_schools if params[:school_search]
-    if params[:slug] == "teaching-subject-now" && !current_claim.eligibility.eligible_itt_subject
+    if new_claim_with_teacher_id?
+      claim_path(current_policy_routing_name, "teacher-detail")
+    elsif params[:slug] == "current-school" && failed_details_check_with_teacher_id?
+      return redirect_to reset_claim_path
+    elsif params[:slug] == "teacher-detail"
+      update_session_with_current_slug
+      return redirect_to claim_path(current_policy_routing_name, next_slug) unless session[:user_info].present?
```

Comment:
We don't need these 2 lines...
```
update_session_with_current_slug
return redirect_to claim_path(current_policy_routing_name, next_slug) unless session[:user_info].present?
```

`set_teacher_detail` seems to be idempotent.

If we remove these 2 lines if the user clicks Yes, they have the option to go "Back" and select No.

At the moment if you select "Yes" you can't go "Back" and select No.